### ISSUE TITLE
RegisterRawNuiCallback Crash

### ISSUE THREAD

**Issue by someone005** [citizenfx/fivem#2783]
### What happened?

I found out that if you will pass empty string to callback function, it will cause game crash.

[Video](https://streamable.com/9rwkt9)

LUA Code:
```lua
RegisterRawNuiCallback('exampleCallback', function(rawData, cb)
    cb('')
end)
```

NUI JS Code
```lua
fetch(`https://${GetParentResourceName()}/exampleCallback`, {
    method: 'POST',
    headers: {
        'Content-Type': 'application/json; charset=UTF-8',
    },
    body: ""
}).catch(err => {})
```

### Expected result

Callback data should be properly packed

### Reproduction steps

1. Create resource with client lua script (description code) and JS Code implemented in script's NUI
2. Start the resource

### Importancy

Crash

### Area(s)

Natives

### Specific version(s)

FiveM: 9837 Server: 9481 Windows

**Comment by AvarianKnight** [comment]
It should be noted that this will happen if you return *any* invalid data during *any* of this, this entire section expects the end user to never send any invalid data via the callback.

**Comment by someone005** [comment]
Yeah, it should noted, but there's one thing. Why the same data passed to RegisterNUICallback cb function doesn't crash the game?

**Comment by AvarianKnight** [comment]
This is a "raw callback" you're expected to return an empty table if you don't want to return anything, or a table with this kind of structure

```cs
struct RequestCallback {
	// optional, doesn't have to be set
	// headers to set for the call back
	Array<(string, string)> headers;

	// alternate to returning an array
	IDictionary<string, string> headers;

	// optional, doesn't have to be set
	// the status code 
	int status;

	// optional, doesn't have to be set
	// the data to return
	string body;
}
```

**Comment by AvarianKnight** [comment]
Here's a patch [patch](https://github.com/user-attachments/files/16964345/rawui.patch) I wrote up yesterday, though its not the cleanest, nor is it tested.
